assets/app/data/crowbar/services/upgrade-status.factory.js

```diff
@@ -0,0 +1,60 @@
+(function() {
+
+    angular
+        .module('suseData.crowbar')
+        .factory('upgradeStatusFactory', upgradeStatusFactory);
+
+    upgradeStatusFactory.$inject = ['$http', '$timeout', 'upgradeFactory'];
+    /* @ngInject */
+    function upgradeStatusFactory($http, $timeout, upgradeFactory) {
+        var factory = {
+            waitForStepToEnd: waitForStepToEnd,
+        };
+
+        return factory;
+
+        /**
+         * Polls for upgrade status until step `step` is `passed`.
+         *
+         * Polling is done at given `pollingInterval` interval.
+         * If waiting finished successfully, `onSuccess` callback is called with last response from status API.
+         * If status API returns error `onError` callback is called with error response and polling is stopped.
+         * If `allowedDowntimeLeft` is specified, temporary unavailability of status API will not trigger
+         * `onError` handler and will not stop polling. The downtime allowance is common for whole call so if
+         * there are multiple short unavailability periods, the total time (sum) is checked.
+         */
+        function waitForStepToEnd(step, onSuccess, onError, pollingInterval, allowedDowntimeLeft) {
+            allowedDowntimeLeft = angular.isDefined(allowedDowntimeLeft) ? allowedDowntimeLeft : 0;
+            upgradeFactory.getStatus()
+                .then(
+                    function (response) {
+                        if (response.data.steps[step].status == 'passed') {
```

Comment:
Mmmh, I dont think that we should change it then. Doing acrobatics with the code just to use the constant in one place seems too much.